### PR TITLE
Add GitHub Action to run `rustfmt`

### PR DESCRIPTION
**Pull request by nasso**

### CODE REVIEW

File: .github/workflows/rustfmt.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install nightly toolchain
+        run: rustup toolchain install nightly
+      - name: Install nightly rustfmt
+        run: rustup component add rustfmt --toolchain nightly
+      - name: Run rustfmt
+        run: cargo +nightly fmt --all -- --check
```
Comment (tronical) [inline .github/workflows/rustfmt.yml:16]: Wouldn't it perhaps make more sense to add this to the existing `ci.yml`? :)

Or, or as long as there are also direct pushes to master then perhaps it makes sense to have both. A check on pull requests and this for direct pushes.

**Comment by tronical** [comment]
I'm all in favour of rustfmt through CI, btw :)

File: .github/workflows/rustfmt.yml (hunk 1 above)
Comment (nasso) [reply .github/workflows/rustfmt.yml:16]: Yeah I wasn't sure where and when this should run. I can merge it into the existing `ci.yml`!

**Comment by tronical** [comment]
Looks good it me! @adamnemecek this ok from your maintainer perspective? (Enforce fmt for PR)

**Comment by adamnemecek** [comment]
Yeah we talked about this on discord. Do you want to squash your commits @nasso ?

**Comment by nasso** [comment]
I don't mind!

**Comment by adamnemecek** [comment]
Yeah.

**Comment by nasso** [comment]
Should be good to go!

**Pull request merged**